### PR TITLE
LyricWiki php fixes

### PR DESCRIPTION
**Pull request by SeanColombo**
Fixes LyricWiki code issues to kill warnings/errors and deprecations related to upgrading to PHP 5.6.

LYR-231, LYR-232, LYR-233

### CODE REVIEW

File: extensions/3rdparty/LyricWiki/server.php [hunk 1]
```diff
@@ -2181,10 +2181,10 @@ function lw_getTitle($artist, $song='', $applyUnicode=true, $allowAllCaps=true){
...
-	$title = preg_replace('/( \()[\']([a-z])/ei', '"$1".strtoupper("\'$2")', $title); // single-quotes like above, but this is needed to avoid escaping the single-quote here.
-	$title = preg_replace('/ [\']([a-z])/ei', '" ".strtoupper("\'$1")', $title); // single-quotes like above, but this is needed to avoid escaping the single-quote here.
+	$title = preg_replace_callback('/([-\("\.\/:_])([a-z])/', function($m){ return $m[1].strtoupper($m[2]); }, $title);
+	$title = preg_replace_callback('/\b(O)[\']([a-z])/i', function($m){ return $m[1].strtoupper("'".$m[2]); }, $title); // single-quotes like above, but this is needed to avoid escaping the single-quote here.  Does it to "O'Riley" but not "I'm" or "Don't"
+	$title = preg_replace_callback('/( \()[\']([a-z])/i', function($m){ return $m[1].strtoupper("'".$m[2]); }, $title); // single-quotes like above, but this is needed to avoid escaping the single-quote here.
+	$title = preg_replace_callback('/ [\']([a-z])/i', function($m){ return " ".strtoupper("'".$m[1]); }, $title); // single-quotes like above, but this is needed to avoid escaping the single-quote here.
```
Comment (macbre) [inline extensions/3rdparty/LyricWiki/server.php]: This code should be unit tested :)

File: extensions/3rdparty/LyricWiki/server.php [hunk 2]
```diff
@@ -2141,7 +2141,7 @@ function lw_fmtAlbum($album,$year){
 ////
 function lw_fmtSong($song){
 	$retVal = rawurldecode(ucwords($song));
-	$retVal = preg_replace('/([-\("\.])([a-z])/e', '"$1".strtoupper("$2")', $retVal);
+	$retVal = preg_replace_callback('/([-\("\.])([a-z])/', function($m){ return $m[1].strtoupper($m[2]); }, $retVal);
 	$retVal = str_replace(" ", "_", $retVal);
 	return $retVal;
 } // end lw_fmtSong()
```
Comment (macbre) [inline extensions/3rdparty/LyricWiki/server.php]: I think it's a good moment to add unit tests here.

Comment (macbre) [inline extensions/3rdparty/LyricWiki/server.php]: @SeanColombo - ping :)

Comment (macbre) [inline extensions/3rdparty/LyricWiki/server.php]: Ping

**Comment by SeanColombo** [comment]
@macbre added unit tests :) ...interestingly, the jenkins tests are failing on a different test related to image-cropping.

I assume that's because a URL or something has changed in "dev" since I branched this. Rebasing (this is a serious rebase!)

File: extensions/3rdparty/LyricWiki/tests/ServerTest.php [hunk 3]
```diff
@@ -67,4 +67,30 @@ public function test_lw_getSearchResults() {
...
+	}
+
+	/**
+	 * @covers lw_fmtSong
+	 */
+	public function test_lw_fmtSong(){
+		$this->assertEquals(lw_fmtSong("guitar%20man"), "Guitar_Man", "Testing URL decoding");
+		$this->assertEquals(lw_fmtSong("guitar (man)"), "Guitar_(Man)", "Testing various capitalization needs.");
+	}
+}
```
Comment (macbre) [inline extensions/3rdparty/LyricWiki/tests/ServerTest.php]: Thanks!

**Comment by macbre** [comment]
LGTM

**Comment by SeanColombo** [comment]
@macbre well, the rebasing didn't seem to help the unit-test issues. Can you still merge it easily?

**Comment by macbre** [comment]
Please cherry pick your commits to the new branch.

**Comment by SeanColombo** [comment]
@macbre Do you know of a way to find out what they were?  The way git does rebasing, the log is now completely garbled with other commits & I don't see a way to filter them out.  I can hard-reset back to a state before the rebase, but I'm not aware of anywhere I can actually see a list of my commits.

**Comment by macbre** [comment]
Did just simply perform `git merge dev` on your branch?

**Pull request closed without merging**